### ISSUE TITLE
Add option to use type_check during struct generation.

### ISSUE THREAD

**Issue by rparcus** [elixir-protobuf/protobuf#355]
TL;DR Would anyone else be interested in having this library optionally generating `@type!` attributes with [type_check](https://github.com/Qqwy/elixir-type_check)?

Benefits of doing that would include:
- being able to use type_check's `@type!` and `@spec!` directly on ProtoBuf generated structs without having to manually create overrides for them (this is a limitation of how type_check works)
- being able to use the StreamData generator facility provided by type_check.

Please let me know if I'm missing something, for example, there could be another way to achieve similar benefits without using type_check.

**Comment by whatyouhide** [comment]
@rparcus enforcing the type_check dependency on users of this library is not something we're planning. This library should expose enough reflection information (through the generated `__message_props__/0` function) that you might be able to do this yourself *on top* of the library 🙃 Let me know if you have questions on this.

Thanks for the proposal anyways!

**Comment by rparcus** [comment]
Thanks @whatyouhide, `__ __message_props__/0` is exactly what I needed 👍
